# Ejemplo de limpieza en pandas 


In [ ]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')


In [ ]:
[3297269,
 6504609,
 7530382,
 15895374,
 17412049,
 17591769,
 18435275,
 19252859,
 20438765,
 23738879,
 28814071,
 28837359,
 38233613,
 40315034,
 41783045,
 65496684,
 79245518,
 79314863]

In [2]:
df = pd.read_excel('datos_CAIMEDv7.xlsx')
df

individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
0       70033505                     2023-08-31  1951-09-10 00:00:00   
1       41760470                     2023-08-30  1955-11-22 00:00:00   
2       17171452                     2023-08-30  1945-11-09 00:00:00   
3     1019032057                     2023-08-25  1989-12-03 00:00:00   
4       19250084                     2023-08-24  1954-12-01 00:00:00   
...          ...                            ...                  ...   
1002     7561381                     2023-05-17  1970-07-09 00:00:00   
1003    41899611                     2023-05-17  1954-11-10 00:00:00   
1004    19317507                     2023-05-17  1956-11-05 00:00:00   
1005     4398004                     2023-05-16  1942-01-24 00:00:00   
1006    41557614                     2023-05-16  1951-05-02 00:00:00   

      edad_calculada estado_civil       sexo lugar_de_nacimiento  \
0               71.0       casado  masculino              bogota   
1               67.0   divorciado   femenino              bogota   
2               77.0       casado  masculino               jesus   
3               33.0  union libre   femenino        barranquilla   
4               68.0       casado  masculino            medellin   
...              ...          ...        ...                 ...   
1002            52.0      soltero  masculino             armenia   
1003            68.0  union libre   femenino             armenia   
1004            66.0       casado  masculino              bogota   
1005            81.0       casado  masculino          montenegro   
1006            72.0       casado   femenino            guateque   

     lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
0                  bogota            profesional                    6.0  ...   
1                  bogota            profesional                    3.0  ...   
2                  bogota           especialista                    4.0  ...   
3                  bogota              tecnologo                    4.0  ...   
4                  bogota            profesional                    6.0  ...   
...                   ...                    ...                    ...  ...   
1002              armenia    secundaria completa                    2.0  ...   
1003              armenia    primaria incompleta                    2.0  ...   
1004             circasia           especialista                    3.0  ...   
1005              armenia      primaria completa                    4.0  ...   
1006              armenia  secundaria incompleta                    3.0  ...   

     Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
0               no               si                                   si   
1               no               si                                   no   
2               no               si                                   no   
3               no               si                                   si   
4               no               si                                   si   
...            ...              ...                                  ...   
1002            no               no                                   no   
1003            no               no                                   no   
1004            no               no                                   no   
1005            si               no                                   no   
1006            si               no                                   no   

     ¿son_signficativos_relacionado_a_obesidad?  \
0                                            si   
1                                            no   
2                                            no   
3                                            si   
4                                            si   
...                                         ...   
1002                                         no   
1003                                       

- **IMC**: Se utilizará la columna de **IMC calculada** (resaltada en verde, basada en valores de peso y talla en la base de datos).
    - Se excluyen del análisis los registros con diferencias en IMC mayores de 2 o menores de -2:
      - ID: `20438765`, `17591769`, `41783045`, `28837359`, `3297269`, `79314863`.

In [3]:
sub_temp = []
index_drop = []
for Id in [20438765, 17591769, 41783045, 28837359, 3297269, 79314863]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result


Indices para Id 20438765: [121]
Indices para Id 17591769: [161]
Indices para Id 41783045: [9]
Indices para Id 28837359: [518]
Indices para Id 3297269: [158]
Indices para Id 79314863: [132]


individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
121   20438765                     2023-05-17  1950-01-08 00:00:00   
161   17591769                     2023-08-10  1977-02-04 00:00:00   
9     41783045                     2023-08-18  1959-10-27 00:00:00   
518   28837359                     2023-07-07  1957-06-22 00:00:00   
158    3297269                     2023-08-17  1949-03-25 00:00:00   
132   79314863                     2023-05-17  1963-09-22 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
121            73.0      soltero   femenino             caqueza   
161            46.0  union libre  masculino              puerto   
9              63.0      soltero   femenino              bogota   
518            66.0        viudo   femenino           mariquita   
158            74.0        viudo  masculino              socota   
132            59.0  union libre   femenino              bogota   

    lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
121               bogota  secundaria incompleta                    3.0  ...   
161                  paz  secundaria incompleta                    2.0  ...   
9                 bogota              tecnologo                    3.0  ...   
518            mariquita    primaria incompleta                    1.0  ...   
158                yopal    primaria incompleta                    1.0  ...   
132               bogota    secundaria completa                    2.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
121            no               no                                   no   
161            no               no                                   no   
9              no               si                                   si   
518            no               no                                   no   
158            no               no                                   no   
132            no               si                                   no   

    ¿son_signficativos_relacionado_a_obesidad?  \
121                                         no   
161                                         no   
9                                           si   
518                                         no   
158                                         no   
132                                         no   

     ¿resultados_relacionados_a_problemas cardiacos?  \
121                                               no   
161                                               no   
9                                                 no   
518                                               no   
158                                               no   
132                                               no   

     ¿son_signficativos_relacionado_a_problemas cardiacos?  \
121                                                 no       
161                                                 no       
9                                                   no       
518                                                 no       
158                                                 no       
132                                                 no       

     ¿resultados_relacionados_a_tension arterial?  \
121                                            no   
161                                            no   
9                                              no   
518                                            no   
158                                            no   
132                                            no   

     ¿son_signficativos_relacionado_a_tension arterial?  \
121                                                 no    
161                                                 no    
9                                                   no    
518                                                 no    
158                                                 no    
132                                                 no    


In [4]:
df.reset_index(drop=True)

individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
0       70033505                     2023-08-31  1951-09-10 00:00:00   
1       41760470                     2023-08-30  1955-11-22 00:00:00   
2       17171452                     2023-08-30  1945-11-09 00:00:00   
3     1019032057                     2023-08-25  1989-12-03 00:00:00   
4       19250084                     2023-08-24  1954-12-01 00:00:00   
...          ...                            ...                  ...   
996      7561381                     2023-05-17  1970-07-09 00:00:00   
997     41899611                     2023-05-17  1954-11-10 00:00:00   
998     19317507                     2023-05-17  1956-11-05 00:00:00   
999      4398004                     2023-05-16  1942-01-24 00:00:00   
1000    41557614                     2023-05-16  1951-05-02 00:00:00   

      edad_calculada estado_civil       sexo lugar_de_nacimiento  \
0               71.0       casado  masculino              bogota   
1               67.0   divorciado   femenino              bogota   
2               77.0       casado  masculino               jesus   
3               33.0  union libre   femenino        barranquilla   
4               68.0       casado  masculino            medellin   
...              ...          ...        ...                 ...   
996             52.0      soltero  masculino             armenia   
997             68.0  union libre   femenino             armenia   
998             66.0       casado  masculino              bogota   
999             81.0       casado  masculino          montenegro   
1000            72.0       casado   femenino            guateque   

     lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
0                  bogota            profesional                    6.0  ...   
1                  bogota            profesional                    3.0  ...   
2                  bogota           especialista                    4.0  ...   
3                  bogota              tecnologo                    4.0  ...   
4                  bogota            profesional                    6.0  ...   
...                   ...                    ...                    ...  ...   
996               armenia    secundaria completa                    2.0  ...   
997               armenia    primaria incompleta                    2.0  ...   
998              circasia           especialista                    3.0  ...   
999               armenia      primaria completa                    4.0  ...   
1000              armenia  secundaria incompleta                    3.0  ...   

     Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
0               no               si                                   si   
1               no               si                                   no   
2               no               si                                   no   
3               no               si                                   si   
4               no               si                                   si   
...            ...              ...                                  ...   
996             no               no                                   no   
997             no               no                                   no   
998             no               no                                   no   
999             si               no                                   no   
1000            si               no                                   no   

     ¿son_signficativos_relacionado_a_obesidad?  \
0                                            si   
1                                            no   
2                                            no   
3                                            si   
4                                            si   
...                                         ...   
996                                          no   
997                                        

- **Edad**: Se utilizará la columna de **edad calculada** (resaltada en verde).
    - Para los tres registros con diferencias en la edad calculada, la **fecha de nacimiento fue verificada** en la base de datos de vacunación (PAIWEB).

- **Presión Arterial Sistólica y Diastólica**: 
    - Se excluyen del análisis los registros en los que la diferencia entre presión arterial sistólica y diastólica es inferior a 20 o superior a 90.
      - ID: `23738879`, `19252859`, `18435275`, `15895374`, `28814071`, `6504609`, `38233613`, `4000000` (resaltados en rojo).

In [5]:
sub_temp = []
index_drop = []
for Id in [23738879, 19252859, 18435275, 15895374, 28814071, 6504609, 38233613, 4000000]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 23738879: [244]
Indices para Id 19252859: [282]
Indices para Id 18435275: [927]
Indices para Id 15895374: [948]
Indices para Id 28814071: [534]
Indices para Id 6504609: [314]
Indices para Id 38233613: [524]
Indices para Id 4000000: [635]


individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
244   23738879                     2023-06-10  1943-11-05 00:00:00   
282   19252859                     2023-05-24  1954-10-23 00:00:00   
927   18435275                     2023-06-06  1972-01-01 00:00:00   
948   15895374                     2023-06-03  1950-01-16 00:00:00   
534   28814071                     2023-07-04  1954-03-31 00:00:00   
314    6504609                     2023-08-15  1946-01-17 00:00:00   
524   38233613                     2023-07-10  1953-10-06 00:00:00   
635    4000000                     2023-06-06                  NaT   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
244            79.0        viudo   femenino              tamara   
282            68.0  union libre  masculino       villavicencio   
927            51.0       casado  masculino             armenia   
948            73.0       casado  masculino              libano   
534            69.0   divorciado   femenino              libano   
314            77.0       casado  masculino            el tambo   
524            69.0       casado   femenino              madrid   
635             NaN          NaN        NaN                 NaN   

    lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
244                yopal  secundaria incompleta                    2.0  ...   
282                yopal  secundaria incompleta                    1.0  ...   
927              armenia    secundaria completa                    2.0  ...   
948              armenia  secundaria incompleta                    3.0  ...   
534               ibague    primaria incompleta                    3.0  ...   
314              acacias    primaria incompleta                    2.0  ...   
524               ibague      primaria completa                    2.0  ...   
635               ibague                    NaN                    NaN  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
244            no               no                                   no   
282            si               si                                   no   
927            no               si                                   si   
948            no               no                                   no   
534            si               no                                   no   
314            no               no                                   no   
524            no               si                                   no   
635            no              NaN                                  NaN   

    ¿son_signficativos_relacionado_a_obesidad?  \
244                                         no   
282                                         no   
927                                         no   
948                                         no   
534                                         no   
314                                         no   
524                                         no   
635                                        NaN   

     ¿resultados_relacionados_a_problemas cardiacos?  \
244                                               no   
282                                               no   
927                                               no   
948                                               no   
534                                               no   
314                                               no   
524                                               no   
635                                              NaN   

     ¿son_signficativos_relacionado_a_problemas cardiacos?  \
244                                                 no       
282                                                 no       
927                                                 no       
948                                                 no       
534                                                 no       
314                                                 no

In [6]:
df.reset_index(drop=True)

individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
0      70033505                     2023-08-31  1951-09-10 00:00:00   
1      41760470                     2023-08-30  1955-11-22 00:00:00   
2      17171452                     2023-08-30  1945-11-09 00:00:00   
3    1019032057                     2023-08-25  1989-12-03 00:00:00   
4      19250084                     2023-08-24  1954-12-01 00:00:00   
..          ...                            ...                  ...   
988     7561381                     2023-05-17  1970-07-09 00:00:00   
989    41899611                     2023-05-17  1954-11-10 00:00:00   
990    19317507                     2023-05-17  1956-11-05 00:00:00   
991     4398004                     2023-05-16  1942-01-24 00:00:00   
992    41557614                     2023-05-16  1951-05-02 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
0              71.0       casado  masculino              bogota   
1              67.0   divorciado   femenino              bogota   
2              77.0       casado  masculino               jesus   
3              33.0  union libre   femenino        barranquilla   
4              68.0       casado  masculino            medellin   
..              ...          ...        ...                 ...   
988            52.0      soltero  masculino             armenia   
989            68.0  union libre   femenino             armenia   
990            66.0       casado  masculino              bogota   
991            81.0       casado  masculino          montenegro   
992            72.0       casado   femenino            guateque   

    lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
0                 bogota            profesional                    6.0  ...   
1                 bogota            profesional                    3.0  ...   
2                 bogota           especialista                    4.0  ...   
3                 bogota              tecnologo                    4.0  ...   
4                 bogota            profesional                    6.0  ...   
..                   ...                    ...                    ...  ...   
988              armenia    secundaria completa                    2.0  ...   
989              armenia    primaria incompleta                    2.0  ...   
990             circasia           especialista                    3.0  ...   
991              armenia      primaria completa                    4.0  ...   
992              armenia  secundaria incompleta                    3.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
0              no               si                                   si   
1              no               si                                   no   
2              no               si                                   no   
3              no               si                                   si   
4              no               si                                   si   
..            ...              ...                                  ...   
988            no               no                                   no   
989            no               no                                   no   
990            no               no                                   no   
991            si               no                                   no   
992            si               no                                   no   

    ¿son_signficativos_relacionado_a_obesidad?  \
0                                           si   
1                                           no   
2                                           no   
3                                           si   
4                                           si   
..                                         ...   
988                                         no   
989                                         no   
990                                         no  

- **cLDL (mg/dL)**: 
    - Tres registros con valores de `2` y `6` fueron excluidos del análisis.
      - ID: `19062972`, `65496684`, `7530382` (resaltados en rojo).

In [7]:
sub_temp = []
index_drop = []
for Id in [19062972, 65496684, 7530382]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 19062972: [858]
Indices para Id 65496684: [19]
Indices para Id 7530382: [705]


individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
858   19062972                     2023-07-06  1946-05-23 00:00:00   
19    65496684                     2023-08-02  1960-03-11 00:00:00   
705    7530382                     2023-12-04  1959-10-21 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
858            77.0      soltero  masculino            quimbaya   
19             63.0      soltero   femenino              armero   
705            64.0  union libre  masculino             pereira   

    lugar_de_procedencia          escolaridad  estrato_socieconomico  ...  \
858             quimbaya  primaria incompleta                    3.0  ...   
19                bogota          profesional                    4.0  ...   
705              armenia    primaria completa                    2.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
858            si               si                                   si   
19             no               si                                   si   
705            no               si                                   si   

    ¿son_signficativos_relacionado_a_obesidad?  \
858                                         no   
19                                          si   
705                                         si   

     ¿resultados_relacionados_a_problemas cardiacos?  \
858                                               no   
19                                                no   
705                                               no   

     ¿son_signficativos_relacionado_a_problemas cardiacos?  \
858                                                 no       
19                                                  no       
705                                                 no       

     ¿resultados_relacionados_a_tension arterial?  \
858                                            no   
19                                             no   
705                                            si   

     ¿son_signficativos_relacionado_a_tension arterial?  \
858                                                 no    
19                                                  no    
705                                                 si    

     ¿resultados_relacionados_a_accidente cerebrovascular?  \
858                                                 no       
19                                                  no       
705                                                 no       

     ¿son_signficativos_relacionado_a_accidente cerebrovascular?  
858                                                 no            
19                                                  no            
705                                                 no            

[3 rows x 142 columns]

In [8]:
df.reset_index(drop=True)

individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
0      70033505                     2023-08-31  1951-09-10 00:00:00   
1      41760470                     2023-08-30  1955-11-22 00:00:00   
2      17171452                     2023-08-30  1945-11-09 00:00:00   
3    1019032057                     2023-08-25  1989-12-03 00:00:00   
4      19250084                     2023-08-24  1954-12-01 00:00:00   
..          ...                            ...                  ...   
985     7561381                     2023-05-17  1970-07-09 00:00:00   
986    41899611                     2023-05-17  1954-11-10 00:00:00   
987    19317507                     2023-05-17  1956-11-05 00:00:00   
988     4398004                     2023-05-16  1942-01-24 00:00:00   
989    41557614                     2023-05-16  1951-05-02 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
0              71.0       casado  masculino              bogota   
1              67.0   divorciado   femenino              bogota   
2              77.0       casado  masculino               jesus   
3              33.0  union libre   femenino        barranquilla   
4              68.0       casado  masculino            medellin   
..              ...          ...        ...                 ...   
985            52.0      soltero  masculino             armenia   
986            68.0  union libre   femenino             armenia   
987            66.0       casado  masculino              bogota   
988            81.0       casado  masculino          montenegro   
989            72.0       casado   femenino            guateque   

    lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
0                 bogota            profesional                    6.0  ...   
1                 bogota            profesional                    3.0  ...   
2                 bogota           especialista                    4.0  ...   
3                 bogota              tecnologo                    4.0  ...   
4                 bogota            profesional                    6.0  ...   
..                   ...                    ...                    ...  ...   
985              armenia    secundaria completa                    2.0  ...   
986              armenia    primaria incompleta                    2.0  ...   
987             circasia           especialista                    3.0  ...   
988              armenia      primaria completa                    4.0  ...   
989              armenia  secundaria incompleta                    3.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
0              no               si                                   si   
1              no               si                                   no   
2              no               si                                   no   
3              no               si                                   si   
4              no               si                                   si   
..            ...              ...                                  ...   
985            no               no                                   no   
986            no               no                                   no   
987            no               no                                   no   
988            si               no                                   no   
989            si               no                                   no   

    ¿son_signficativos_relacionado_a_obesidad?  \
0                                           si   
1                                           no   
2                                           no   
3                                           si   
4                                           si   
..                                         ...   
985                                         no   
986                                         no   
987                                         no  

- **Colesterol**: 
    - Se realizó una validación del reporte de colesterol calculando la diferencia entre el colesterol total y la suma de colesterol HDL + LDL.
    - Se encontraron 9 registros sin datos de colesterol y 3 registros en los que el valor era igual o menor a cero, por lo que fueron excluidos del análisis.
      - ID: `79245518`, `17412049`, `40315034` (resaltados en rojo).

In [9]:
sub_temp = []
index_drop = []
for Id in [79245518, 17412049, 40315034]:
    rows = df[df['individuo'] == Id]
    indices = rows.index.tolist()
    index_drop.append(indices)
    sub_temp.append(rows)
    df= df.drop(int(indices[0]))
    print(f"Indices para Id {Id}: {indices}")
result = pd.concat(sub_temp, ignore_index=False)
result

Indices para Id 79245518: [49]
Indices para Id 17412049: [302]
Indices para Id 40315034: [323]


individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
49    79245518                     2023-06-23  1969-07-15 00:00:00   
302   17412049                     2023-09-01  1963-06-18 00:00:00   
323   40315034                     2023-07-28  1957-03-16 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
49             53.0       casado  masculino              bogota   
302            60.0      soltero  masculino             acacias   
323            66.0        viudo   femenino             yarumal   

    lugar_de_procedencia             escolaridad  estrato_socieconomico  ...  \
49                bogota             profesional                    5.0  ...   
302              acacias  pre-escolar incompleta                    1.0  ...   
323              acacias     primaria incompleta                    2.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
49             no               si                                   si   
302            si               no                                   no   
323            si               no                                   no   

    ¿son_signficativos_relacionado_a_obesidad?  \
49                                          si   
302                                         no   
323                                         no   

     ¿resultados_relacionados_a_problemas cardiacos?  \
49                                                no   
302                                               no   
323                                               no   

     ¿son_signficativos_relacionado_a_problemas cardiacos?  \
49                                                  no       
302                                                 no       
323                                                 no       

     ¿resultados_relacionados_a_tension arterial?  \
49                                             no   
302                                            no   
323                                            no   

     ¿son_signficativos_relacionado_a_tension arterial?  \
49                                                  no    
302                                                 no    
323                                                 no    

     ¿resultados_relacionados_a_accidente cerebrovascular?  \
49                                                  no       
302                                                 no       
323                                                 no       

     ¿son_signficativos_relacionado_a_accidente cerebrovascular?  
49                                                  no            
302                                                 no            
323                                                 no            

[3 rows x 142 columns]

## Datos luego de extracciones:
    Finalmete se mantienen 987 regsitros 

In [10]:
df = df.reset_index(drop=True)
df

individuo fecha_constetimiento_informado  fecha_de_nacimiento  \
0      70033505                     2023-08-31  1951-09-10 00:00:00   
1      41760470                     2023-08-30  1955-11-22 00:00:00   
2      17171452                     2023-08-30  1945-11-09 00:00:00   
3    1019032057                     2023-08-25  1989-12-03 00:00:00   
4      19250084                     2023-08-24  1954-12-01 00:00:00   
..          ...                            ...                  ...   
982     7561381                     2023-05-17  1970-07-09 00:00:00   
983    41899611                     2023-05-17  1954-11-10 00:00:00   
984    19317507                     2023-05-17  1956-11-05 00:00:00   
985     4398004                     2023-05-16  1942-01-24 00:00:00   
986    41557614                     2023-05-16  1951-05-02 00:00:00   

     edad_calculada estado_civil       sexo lugar_de_nacimiento  \
0              71.0       casado  masculino              bogota   
1              67.0   divorciado   femenino              bogota   
2              77.0       casado  masculino               jesus   
3              33.0  union libre   femenino        barranquilla   
4              68.0       casado  masculino            medellin   
..              ...          ...        ...                 ...   
982            52.0      soltero  masculino             armenia   
983            68.0  union libre   femenino             armenia   
984            66.0       casado  masculino              bogota   
985            81.0       casado  masculino          montenegro   
986            72.0       casado   femenino            guateque   

    lugar_de_procedencia            escolaridad  estrato_socieconomico  ...  \
0                 bogota            profesional                    6.0  ...   
1                 bogota            profesional                    3.0  ...   
2                 bogota           especialista                    4.0  ...   
3                 bogota              tecnologo                    4.0  ...   
4                 bogota            profesional                    6.0  ...   
..                   ...                    ...                    ...  ...   
982              armenia    secundaria completa                    2.0  ...   
983              armenia    primaria incompleta                    2.0  ...   
984             circasia           especialista                    3.0  ...   
985              armenia      primaria completa                    4.0  ...   
986              armenia  secundaria incompleta                    3.0  ...   

    Antidiabético ¿se_hizo_examen? ¿resultados_relacionados_a_obesidad?  \
0              no               si                                   si   
1              no               si                                   no   
2              no               si                                   no   
3              no               si                                   si   
4              no               si                                   si   
..            ...              ...                                  ...   
982            no               no                                   no   
983            no               no                                   no   
984            no               no                                   no   
985            si               no                                   no   
986            si               no                                   no   

    ¿son_signficativos_relacionado_a_obesidad?  \
0                                           si   
1                                           no   
2                                           no   
3                                           si   
4                                           si   
..                                         ...   
982                                         no   
983                                         no   
984                                         no  